src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -499,6 +499,9 @@ private async Task RestoreAsync(bool forceRestore, RestoreOperationSource restor
                                 {
                                     if (isRestoreSucceeded)
                                     {
+                                        var errors = restoreSummaries.SelectMany(summary => summary.Errors.Where(e => e.Code.Equals(NuGetLogCode.NU1901) || e.Code.Equals(NuGetLogCode.NU1902) || e.Code.Equals(NuGetLogCode.NU1903) || e.Code.Equals(NuGetLogCode.NU1904)));
```

Comment:
We're iterating over *all* warnings/errors here, can we just stop at the first one? 
It will be a lot more efficient.

Also given that this is done every time, I think we can just use for loops instead of linq or foreach and avoid allocations.

Reply:
Just a reminder to use a for loop here and stop at the first issue.